core/Pos/Core/Script.hs

```diff
@@ -22,56 +22,52 @@ import           Pos.Binary.Class   (Bi, getCopyBi, putCopyBi)
 ----------------------------------------------------------------------------
 
 deriving instance Show PLCore.PrimData
+deriving instance Show PLCore.SimplePattern
 
 deriving instance (Show a, Show (PLCore.ClauseF a)) => Show (PLCore.TermF a)
 deriving instance (Eq a, Eq (PLCore.ClauseF a)) => Eq (PLCore.TermF a)
 
 deriving instance Eq PLTypes.TyConSig
 deriving instance Eq PLTypes.ConSig
-deriving instance Eq PLTypes.PolymorphicType
+deriving instance Eq PLCore.SimplePattern
 deriving instance Eq PLCore.Program
 
-deriveEq1 ''PLCore.PatternF
 deriveEq1 ''PLCore.ClauseF
 deriveEq1 ''PLCore.TermF
 deriveEq1 ''PLTypes.TypeF
 
-deriving instance Show (f PLCore.PatternF) => Show (PLCore.PatternF (f PLCore.PatternF))
-
 deriving instance Show r => Show (PLCore.ClauseF r)
 
 instance Store Vars.FreeVar
 instance Store Vars.MetaVar
 instance Store Vars.BoundVar
 instance Store PLTypes.TyConSig
 instance Store PLTypes.ConSig
-instance Store PLTypes.PolymorphicType
 instance Store a => Store (Names.Sourced a)
 instance Store ABT.Variable
 instance Store (f (ABT.Scope f)) => Store (ABT.ABT f)
 instance Store (f (ABT.Scope f)) => Store (ABT.Scope f)
 instance Store r => Store (PLCore.ClauseF r)
 instance Store a => Store (PLCore.TermF a)
-instance Store a => Store (PLCore.PatternF a)
 instance Store a => Store (PLTypes.TypeF a)
 instance Store PLCore.PrimData
+instance Store PLCore.SimplePattern
```

Comment:
This part makes me think that maybe we should wait until #1203 is merged, because it has higher priority.